lib/ddtrace/writer.rb

```diff
@@ -29,53 +31,63 @@ def initialize(options = {})
 
       @traces_flushed = 0
 
-      # spawns two different workers for spans and services;
-      # they share the same transport which is thread-safe
-      @trace_worker = nil
-      @service_worker = nil
+      # one worker for both services and traces, each have their own queues
+      @worker = nil
     end
 
     # spawns two different workers for spans and services;
     # they share the same transport which is thread-safe
     def start
-      @trace_worker = Datadog::Workers::AsyncTransport.new(@span_interval, @transport, @buff_size) do |items, transport|
-        send_spans(items, transport)
-      end
-      @service_worker = Datadog::Workers::AsyncTransport.new(@service_interval, @transport, 1) do |items, transport|
-        send_services(items, transport)
+      @next_send_services = Time.now
```

Comment:
I think it could be better to put this attribute inside the worker. I think that the ``writer`` component should only handle stuff and enqueue jobs in the right buffer.